www/src/pages/insights.jsx

```diff
@@ -43,40 +53,79 @@ const componentsUsage = dependentProjectsUsages.reduce((accumulator, project) =>
-  return {
-    name: componentName,
-    count: componentUsageCounts,
-    usages: componentsUsage[componentName],
-  };
-});
+const round = (n) => Math.round(n * 10) / 10;
+
+const getEmptyMessage = (text) => `Currently there are no ${text} usage yet`;
 
 const SummaryUsage = () => {
+  const { paragonTypes = {}, isParagonIcon = () => false } = useContext(InsightsContext);
+  const isMedium = useMediaQuery({ minWidth: breakpoints.large.minWidth });
+
+  const typeCount = Object.keys(paragonTypes).reduce((accumulator, componentName) => {
+    const type = paragonTypes[componentName] || (isParagonIcon(componentName) && ICON_TYPE);
+    if (componentsInUsage.includes(componentName)) {
+      accumulator[type] = (accumulator[type] || 0) + 1;
+    }
+    return accumulator;
+  }, {});
+
+  const filterValues = Object.keys(paragonTypes)
+    .map((key) => paragonTypes[key])
+    .filter((v, i, a) => a.indexOf(v) === i)
+    .map(type => ({ name: type, number: typeCount[type], value: type }));
+  // Number of Icons is calculated in the statement below. Initialized as `undefined` to not display '0'.
+  const iconsType = { name: ICON_TYPE, number: undefined, value: ICON_TYPE };
+
+  const summaryComponentsUsage = Object.entries(componentsUsage).map(([componentName, usages]) => {
+    const componentUsageCounts = usages.reduce((accumulator, project) => accumulator + project.componentUsageCount, 0);
+    let type = paragonTypes[componentName];
+    if (!type && isParagonIcon(componentName)) {
+      type = ICON_TYPE;
+      iconsType.number = (iconsType.number || 0) + 1;
+    }
+    return {
+      name: componentName,
+      count: componentUsageCounts,
+      usages: componentsUsage[componentName],
+      type,
+    };
+  });
+  filterValues.push(iconsType);
+  typeCount[ICON_TYPE] = iconsType.number;
+
   const summaryTableData = summaryComponentsUsage.sort((a, b) => {
     const nameA = a.name.toUpperCase();
     const nameB = b.name.toUpperCase();
-    return (nameA < nameB) ? -1 : (nameA > nameB) ? 1 : 0;
+    return nameA < nameB ? -1 : 1;
   });
-  const round = (n) => Math.round(n * 10) / 10;
-  const averageComponentsUsedPerProject = dependentProjects.reduce((accumulator, project) => accumulator += project.count, 0) / dependentProjects.length;
+
+  const averageComponentsUsedPerProject = dependentProjects.reduce(
+    (accumulator, project) => accumulator + project.count, 0,
+  ) / dependentProjects.length;
+
   return (
     <div className="pt-5 mb-5">
       <div className="mb-5">
         <h2>Overview</h2>
         <p>
           Paragon is used by at least <strong>{dependentProjects.length} projects</strong>, with an average
-          of <strong>{round(averageComponentsUsedPerProject)} component instances</strong> per project.
+          of <strong>{round(averageComponentsUsedPerProject)} imports</strong> per project.
         </p>
       </div>
-      <h3>Overall component usage</h3>
+      <h3>Overall usage</h3>
       <DataTable
```

Comment:
From the previous review:

> When applying filters in the sidebar, it looks like it's updating the table contents but the count in the status "Showing 20 of 207" doesn't seem to change. For example, when there's a filter of "hooks" with just 5 results, I think that message would update to "Showing 5 of 5"?

I think this is still an outstanding issue? In the screenshot below, I have "Hooks" and "Objects" selected; the entire data set of results is only 5 results with those filters applied, but the table status is saying "Showing 5 **of 207**" instead of "Showing 5 **of 5**".

As is, the "Showing 5 of 207" hints there should be table pagination, but with those filters applied there isn't 😄 

<img width="981" alt="image" src="https://user-images.githubusercontent.com/2828721/185612963-542ca6cd-2188-4a19-a2be-a0b259a4fac6.png">

Reply:
Hm, I agree with this. I went to check how it works in examples in the `DataTable` and seems it is default behavior for filtering. As for me it is also correct to display `Showing 5 of 5.` instead of `of 207.`
![image](https://user-images.githubusercontent.com/36768631/185912663-ca9c3cac-a316-4c2f-988b-54df683c5f4c.png)
![image](https://user-images.githubusercontent.com/36768631/185912984-f09a8a71-79ca-40bc-987c-cac14141e6d9.png)
I think meaning of `Showing 5 of 207.` is perfect when `isPaginated` is not used. But with pagination we can not see how much records per chosen filters. So maybe it is good idea to change behavior for this label when `isPaginated` is passed in `DataTable` component. What do you think?